state/state_test.go

```diff
@@ -1019,3 +1030,36 @@ func TestApplyUpdates(t *testing.T) {
 		assert.Equal(t, tc.expected, res, "case %d", i)
 	}
 }
+
+func TestStateProto(t *testing.T) {
+	tearDown, _, state := setupTestCase(t)
+	defer tearDown(t)
+
+	tc := []struct {
+		testName string
+		state    *sm.State
+		expPass  bool
+	}{
+		{"empty state", &sm.State{}, false},
+		{"nil failure state", nil, false},
+		{"success state", &state, true},
+	}
+
+	for _, tt := range tc {
+		tt := tt
+		pbs, err := tt.state.ToProto()
```

Comment:
would be cool to use `testing/quick` package here

Reply:
mhh, I can take a look at a later time. It would be nice to be consistent in how we test things tho